app/assets/javascripts/components/user_photo_edit.es6.jsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * @prop user - user
+ * @prop images - photo gallery for user
+**/
+class UserPhotoEdit extends React.Component {
+constructor(props) {
+    super(props);
+    this._openModal = this._openModal.bind(this);
+    this._cancelEdit = this._cancelEdit.bind(this);
+    this._enableEdit = this._enableEdit.bind(this);
+    this._disableEdit = this._disableEdit.bind(this);
+    this._selectImage = this._selectImage.bind(this);
+    this._handleFileChange = this._handleFileChange.bind(this);
+    this._handleTextChange = this._handleTextChange.bind(this);
+    this._renderImages = this._renderImages.bind(this);
+    this._saveForm = this._saveForm.bind(this);
+    this._successfulSave = this._successfulSave.bind(this);
+    this.state = {
+      images: this.props.images,
+      photo_ids: [],
```

Comment:
nit: I don't think "photo_ids" or "photo_data" is used anywhere else in this file - make sure all state elements are used or delete them!